lib/cfn-nag/cli_options.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require 'trollop'
+
+# rubocop:disable Metrics/BlockLength
+# rubocop:disable Metrics/AbcSize
+# rubocop:disable Metrics/MethodLength
+def cli_options(cfn_nag_scan: false)
+  options_message = '[options] <cloudformation template path ...>|' \
+                    '<cloudformation template in STDIN>'
+
+  input_path_message = 'CloudFormation template to nag on or directory of ' \
+                       'templates.  Default is all *.json, *.yaml, *.yml ' \
+                       'and *.template recursively, but can be constrained ' \
+                       'by --template-pattern'
+
+  custom_rule_exceptions_message = 'Isolate custom rule exceptions - just ' \
+                                   'emit the exception without stack trace ' \
+                                   'and keep chugging'
+
+  template_pattern_message = 'Within the --input-path, match files to scan ' \
+                             'against this regular expression'
+
+  version = Gem::Specification.find_by_name('cfn-nag').version
+
+  if cfn_nag_scan
+    Trollop.options do
+      version version
+      opt :input_path,
```

Comment:
can we do one set of Trollop.options, but use the "cfn_nag_scan" flag to determine which short arg to force per opt for backward compat?

Reply:
also break the options into common vs. divergent.